src/circuit/circuit_frame.rs

```diff
@@ -3055,6 +3175,198 @@ fn apply_continuation<F: LurkField, CS: ConstraintSystem<F>>(
 
         let res = AllocatedPtr::from_parts(&res_tag, &val);
 
+        // Next constraints are used for number comparisons
+        ///////////////////////////////////////////////////////////////////////
+        let double_a = constraints::add(&mut cs.namespace(|| "double a"), a, a)?;
+        let double_a_bits = double_a
+            .to_bits_le_strict(&mut cs.namespace(|| "double a lsb"))
+            .unwrap();
+        let lsb_a = double_a_bits.get(0);
+        let bool_val_a = match lsb_a {
+            Some(Boolean::Is(v)) => v.get_value(),
+            _ => {
+                Some(false) // Blank
+            }
+        };
+        let a_is_negative = Boolean::from(
+            AllocatedBit::alloc(cs.namespace(|| "boolean val a"), bool_val_a).unwrap(),
+        );
```

Comment:
AllocateBit is unconstrained.

Reply:
GitHub elides too much context here. The lines which create the unconstrained AllocatedBit are just above (from 3185):
```
        let bool_val_a = match lsb_a {
            Some(Boolean::Is(v)) => v.get_value(),
            _ => {
                Some(false) // Blank
            }
        };
```

Reply:
Fixed